// @shopify/cli/dist/local-CWK27QUL.js
import {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  themeToken,
  useEmbeddedThemeCLI,
  useThemebundling
} from "./chunk-3GAW7KHI.js";
import "./chunk-P7JUHQKL.js";
import "./chunk-27ILOE3X.js";
import "./chunk-SNOECVP4.js";
import "./chunk-6XKWVDB6.js";
import "./chunk-UBB7JKND.js";
import "./chunk-T2GATXSP.js";
import "./chunk-XVNW332R.js";
import "./chunk-A3JXD7N7.js";
import "./chunk-7Q3MMWAC.js";
import "./chunk-MX4RJDBM.js";
import "./chunk-NB4NLOEJ.js";
import "./chunk-522OB3EU.js";
import "./chunk-PNN7RS7Y.js";
import "./chunk-POZ5MGPT.js";
export {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  themeToken,
  useEmbeddedThemeCLI,
  useThemebundling
};
